expan/core/results.py

```diff
@@ -0,0 +1,110 @@
+    :param control_statistics: statistics within the control group
+    :type  control_statistics: SampleStatistics
+    :param treatment_statistics: statistics within the treatment group
+    :type  treatment_statistics: SampleStatistics
+    """
+    def __init__(self, control_statistics, treatment_statistics):
+        self.control_statistics   = control_statistics
+        self.treatment_statistics = treatment_statistics
+
+
+class SampleStatistics(object):
+    """ This class holds sample size, mean and variance.
+    :type sample_size: int
+    :type mean: float
+    :type variance: float
+    """
+    def __init__(self, sample_size, mean, variance):
+        self.sample_size = sample_size
+        self.mean        = mean
+        self.variance    = variance
+
+
+class SimpleTestStatistics(BaseTestStatistics):
+    """ Additionally to BaseTestStatistics, holds delta, confidence interval, statistical power, and p value.
+    :type control_statistics: SampleStatistics
+    :type treatment_statistics: SampleStatistics
+    :type delta: float
+    :type p: float
+    :type statistical_power: float
+    :type ci: ConfidenceInterval
+    """
+    def __init__(self, control_statistics, treatment_statistics, delta, ci, p, statistical_power):
+        super(SimpleTestStatistics, self).__init__(control_statistics, treatment_statistics)
+        self.delta               = delta
+        self.p                   = p
+        self.statistical_power   = statistical_power
+        self.confidence_interval = ci
+
+
+class EarlyStoppingTestStatistics(SimpleTestStatistics):
+    """ Additionally to SimpleTestStatistics, holds boolean flag for early stopping.
+    :type control_statistics: SampleStatistics
+    :type treatment_statistics: SampleStatistics
+    :type delta: float
+    :type p: float
+    :type statistical_power: float
+    :type ci: ConfidenceInterval
+    :type stop: bool
+    """
+    def __init__(self, control_statistics, treatment_statistics, delta, ci, p, statistical_power, stop):
+        super(EarlyStoppingTestStatistics, self).__init__(control_statistics, treatment_statistics, delta, ci, p, statistical_power)
+        self.stop = stop
+
+
+class CorrectedTestStatistics(object):
+    """ Holds original and corrected statistics. This class should be used to hold statistics for multiple testing.
+    original_test_statistics and corrected_test_statistics should have the same type.
+    :param original_test_statistics: test result before correction
+    :type  original_test_statistics: SimpleTestStatistics or EarlyStoppingTestStatistics
+    :param corrected_test_statistics: test result after correction
+    :type  corrected_test_statistics: SimpleTestStatistics or EarlyStoppingTestStatistics
+    """
+    def __init__(self, original_test_statistics, corrected_test_statistics):
+        type1 = type(original_test_statistics)
+        type2 = type(corrected_test_statistics)
+        if type1 != type2:
```

Comment:
we could also double check if those are instances of subclasses of `BaseTestStatistics`, i.e. by

`isinstance(original_test_statistics, BaseTestStatistics)`

Reply:
good point!